localstack/services/edge.py

```diff
@@ -400,34 +402,42 @@ def ensure_can_use_sudo():
         run("sudo -v", stdin=True)
 
 
-def start_component(component: str, port=None):
+def start_component(component: str, listen_str: str = None):
     if component == "edge":
-        return start_edge(port=port)
+        return start_edge(listen_str=listen_str)
     if component == "proxy":
-        return start_proxy(port=port)
+        return start_proxy(listen_str=listen_str)
     raise Exception("Unexpected component name '%s' received during start up" % component)
 
 
-def start_proxy(port, asynchronous=False):
+def start_proxy(
+    listen_str: str, target_address: Optional[HostAndPort] = None, asynchronous: bool = False
+) -> FuncThread:
     """
     Starts a TCP proxy to perform a low-level forwarding of incoming requests.
     The proxy's source port (given as method argument) is bound to the EDGE_BIND_HOST.
     The target IP is always 127.0.0.1.
-    The target port is parsed from the EDGE_FORWARD_URL (for compatibility with the legacy edge proxy forwarding).
-    All other parts of the EDGE_FORWARD_URL are _not_ used anymore.
 
     :param port: source port
     :param asynchronous: False if the function should join the proxy thread and block until it terminates.
     :return: created thread executing the proxy
     """
-    destination_port = urlparse(config.EDGE_FORWARD_URL).port
-    if not destination_port or destination_port < 1 or destination_port > 65535:
-        raise ValueError("EDGE_FORWARD_URL does not contain a valid port.")
+    listen_hosts = parse_gateway_listen(listen_str)
+    listen = listen_hosts[0]
+    return do_start_tcp_proxy(listen, target_address, asynchronous)
+
+
+def do_start_tcp_proxy(
+    listen: HostAndPort, target_address: Optional[HostAndPort] = None, asynchronous: bool = False
+) -> FuncThread:
+    if target_address is None:
+        target_address = HostAndPort(host=config.LOCALHOST_IP, port=constants.DEFAULT_PORT_EDGE)
+
+    src = f"{listen.host}:{listen.port}"
+    dst = f"{config.LOCALHOST_IP}:{target_address.port}"
```

Comment:
That's a good point. I think it would be better to use `target_address` here. I can't find a usage of this function that specifies a `target_address` so we are likely always using the default set on L434.